Cue/app/reducers/library.js

```diff
@@ -1,28 +1,127 @@
 // @flow
 
-import type {Action} from '../actions/types';
-import type {Deck} from '../api/types';
+import type { Action } from '../actions/types';
+import type { Deck } from '../api/types';
 
 export type State = {
   decks: ?Array<Deck>;
+  localChanges: [];
 };
 
 const initialState: State = {
   decks: null,
+  localChanges: [],
 };
 
 // if state.decks is null then library hasn't loaded yet
 // if state.decks size is 0, then library is empty
-function library(state: State = initialState , action: Action): State {
-  switch (action.type) {
-    case 'LOADED_LIBRARY':
-      return {
-        decks: action.decks,
-      };
-   	default:
-   		return state;
+function library(state: State = initialState, action: Action): State {
+  let decks: Array<Deck> = state.decks == null ? [] : state.decks.slice()
+  let localChanges = state.localChanges.slice()
+
+  if (action.type === 'LOADED_LIBRARY') {
+    decks = action.decks
+
+  } else if (action.type === 'DECK_CREATED') {
+    decks.push(action.deck)
+    localChanges.push({...action.deck, action: "add"})
+
+  } else if (action.type === 'DECK_DELETED') {
+    let deckIndex = decks.findIndex(deck => deck.uuid == action.uuid)
+    if (deckIndex >= 0)
+      decks.splice(deckIndex,1)
+    let changeIndex = localChanges.findIndex(deck => deck.uuid == action.uuid)
```

Comment:
Is it possible that we could have multiple changes for the same deck in `localChanges`? If so, this would only find the first change I think